### PR TITLE
[GH-327] Add Speed

### PR DESCRIPTION
**Pull request by lotuuu**
## Motivation

Add a new "Speed" stat that affects basic skill cooldowns. We're fine with only having it used on skill cast (instead of doint it retroactively). An issue for this scenario has been created already: https://github.com/lambdaclass/champions_of_mirra/issues/423
Closes https://github.com/lambdaclass/champions_of_mirra/issues/327

Characters don't have a base speed, since it doesn't make sense: it's easier and more balance-able to adjust their cooldowns instead!

## Summary of changes

- Add speed stat calculations in Units (for items)
- Implement speed in skill cooldown calculation
- Implement tests for speed buffs and debuffs
- Gives some love to the Simulator module doc

## How to test it?

Connect from the client, give a unit in your team some skill that messes with speed like:
```json
  {
    "name": "D'Agna Basic",
    "energy_regen": 0,
    "animation_duration": 500,
    "mechanics": [
      {
        "trigger_delay": 250,
        "apply_effects_to": {
          "targeting_strategy": {
            "type": "random",
            "count": 6,
            "target_allies": true
          },
          "effects": [
            {
              "type": {
                "type": "permanent"
              },
              "initial_delay": 0,
              "components": [],
              "modifiers": [
                {
                  "attribute": "speed",
                  "modifier_operation": "Add",
                  "float_magnitude": 500
                }
              ],
              "executions": []
            }
          ]
        }
      }
    ],
    "cooldown": 200
  },
```

And see how all your units start going hypermode thanks to the speed buff after they cast their first basic skill!
note: they will be limited by their animation_duration and trigger_delay. You can set all of those to 0 to see _real_ hypermode ;)

### CODE REVIEW

**Review by ncontinanza — APPROVED** [review]
LGTM!

File: apps/champions/lib/champions/battle/simulator.ex [hunk 1]
```diff
@@ -267,6 +284,16 @@ defmodule Champions.Battle.Simulator do
...
   end
 
+  defp calculate_cooldown(skill, unit) do
+    speed = calculate_unit_stat(unit, :speed) |> Decimal.from_float()
+
+    divisor = Decimal.div(Decimal.max(-99, speed), 100) |> Decimal.add(1)
+
+    Decimal.div(skill.base_cooldown, divisor)
+    |> Decimal.round()
+    |> Decimal.to_integer()
```
[the remark this answers is not in the document]
Comment (lotuuu) [reply apps/champions/lib/champions/battle/simulator.ex:294]: Needs to be rounded first!

https://hexdocs.pm/decimal/Decimal.html#to_integer/1

[the remark this answers is not in the document]
Comment (agustinesco) [reply apps/champions/lib/champions/battle/simulator.ex:294]: nice

**Review by agustinesco — APPROVED** [review]
LGTM! tested adding speed to muflus basic and runned the battle.md script and the battle is quite faster! nice

**Pull request merged**